Build failure: bluej

### Steps To Reproduce

Steps to reproduce the behavior:
1. build bluej at 5.2.0

### Additional context

The build failure occurs within openjfx, but not when i only install openjfx or openjdk itself.

### Build log

Log is too long to copy and does not tell you anything except that it failed during a perl command. I'll try to upload it somehow after I've tried another (the 3rd) rebuild. 

Log is 1.2M and cannot be uploaded to gist (bad request). Here it is from my own host: https://data.tudbut.de/openjfx_log

### Notify maintainers

@chvp 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.50, NixOS, 24.11 (Vicuna), 24.11pre679633.1355a0cbfeac`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

The relevant error is:
```
/build/source/modules/javafx.web/src/main/native/Source/JavaScriptCore/offlineasm/parser.rb:877:in `block in parseIncludes': undefined method `exists?' for class File (NoMethodError)

                if not File.exists?(fileName)
                           ^^^^^^^^
Did you mean?  exist?
	from <internal:kernel>:187:in `loop'
	from /build/source/modules/javafx.web/src/main/native/Source/JavaScriptCore/offlineasm/parser.rb:858:in `parseIncludes'
	from /build/source/modules/javafx.web/src/main/native/Source/JavaScriptCore/offlineasm/parser.rb:914:in `parseHash'
	from /build/source/modules/javafx.web/src/main/native/Source/JavaScriptCore/offlineasm/generate_settings_extractor.rb:54:in `<main>'
make[2]: *** [Source/JavaScriptCore/CMakeFiles/LLIntSettingsExtractor.dir/build.make:104: JavaScriptCore/DerivedSources/LLIntDesiredSettings.h] Error 1
make[1]: *** [CMakeFiles/Makefile2:597: Source/JavaScriptCore/CMakeFiles/LLIntSettingsExtractor.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```
Which is due to the upgrade to Ruby 3 (#224018), it removed this deprecated method. The fix is simple, I will open a PR shortly.

--- a/pkgs/development/compilers/openjdk/openjfx/17/default.nix
+++ b/pkgs/development/compilers/openjdk/openjfx/17/default.nix
@@ -48,6 +48,9 @@ in stdenv.mkDerivation {
       -i modules/javafx.web/src/main/native/Source/bmalloc/bmalloc/Heap.cpp \
          modules/javafx.web/src/main/native/Source/bmalloc/bmalloc/IsoSharedPageInlines.h
 
+    substituteInPlace modules/javafx.web/src/main/native/Source/JavaScriptCore/offlineasm/parser.rb \
+      --replace-fail "File.exists?" "File.exist?"
+
     ln -s $config gradle.properties
   '';
 

